homeassistant/components/tuya/climate.py

```diff
@@ -474,7 +484,11 @@ def swing_mode(self) -> str:
     def turn_on(self) -> None:
         """Turn the device on, retaining current HVAC (if supported)."""
         if DPCode.SWITCH in self.device.function:
-            self._send_command([{"code": DPCode.SWITCH, "value": True}])
+            self._send_command([{"code": DPCode.POWER, "value": True}])
+            return
+
+        if DPCode.POWER_SWITCH in self.device.function:
+            self._send_command([{"code": DPCode.POWER_SWITCH, "value": True}])
```

Comment:
We shouldn't be determining this every action. Instead let's figure this out when the entity is set up, so the right DPCode can be referenced correctly every single time.

../Frenck

Reply:
Hello @frenck , thank you for this review. I will try to do this (this is my first PR in HA)